ufo/ufo-resources.c

```diff
@@ -604,6 +604,8 @@ add_program_from_source (UfoResourcesPrivate *priv,
         build_options = g_string_new (priv->build_opts->str);
         opt_append_device_options (build_options, priv, i);
         opt_append_include_paths (build_options, priv);
+        g_string_append (build_options, " ");
```

Comment:
I know this is not strictly necessary but it's more error-proof since filter developers don't know they should add a space. And an extra free space should be of no issue.

Reply:
In principle you are right but I had problems with the Intel OpenCL compiler which was choking on too many spaces. So best, strip leading and trailing spaces and then add the one in front.